internal/data/postgres/carts/carts.go

```diff
@@ -0,0 +1,84 @@
+package carts
+
+import (
+	"context"
+
+	errors "github.com/Red-Sock/trace-errors"
+	"github.com/jackc/pgx/v5"
+
+	"github.com/Red-Sock/Red-Cart/internal/clients/postgres"
+	"github.com/Red-Sock/Red-Cart/internal/domain/cart"
+)
+
+type Carts struct {
+	conn postgres.Conn
+}
+
+func New(conn postgres.Conn) *Carts {
+	return &Carts{conn: conn}
+}
+
+func (c *Carts) GetByCartId(ctx context.Context, cartId int64) (cart.Cart, error) {
+	var dbCart cart.Cart
+
+	err := c.conn.QueryRow(ctx, `SELECT id, owner_id FROM cart
+	    WHERE id = $1`, cartId).Scan(&dbCart.Id, &dbCart.OwnerId)
+	if errors.Is(err, pgx.ErrNoRows) {
+		return dbCart, nil
+	}
+
+	if err != nil {
+		return cart.Cart{}, errors.Wrap(err, "error getting cart by ownerId from database")
+	}
```

Comment:
одно в другом нужно сделать
```suggestion
	if err != nil {
		if errors.Is(err, pgx.ErrNoRows) {
			return dbCart, nil
		}
		return cart.Cart{}, errors.Wrap(err, "error getting cart by ownerId from database")
	}
```